scipy/stats/tests/common_tests.py

```diff
@@ -335,3 +336,214 @@ def check_rvs_broadcast(distfunc, distname, allargs, shape, shape_only, otype):
         np.random.seed(123)
         expected = rvs(*allargs)
         assert_allclose(sample, expected, rtol=1e-13)
+
+
+def check_deprecation_warning_gh5982_moment(distfn, arg, distname):
```

Comment:
I didn't want to add all these different combinations, but I kept thinking of cases which not behave as desired, so I started enumerating all the possibilities. Ultimately, if we get the logic right in `rv_continuous`, we could be confident that we'll get the behavior we want without testing all the possibilities explicitly, but for now...